### DISCUSSION TITLE
캘린더 기능에 대해

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#205] [뜨거운 토론]
# 캘린더에서 무엇을 보여주는가?

- PetPlantHistory에 대한 정보를 보여준다.
- 캘린더는 리마인더 정보 + History 정보를 보여준다.
- 물주기 기록에 대해 전반적으로 확인할 수 있다고 생각하여 캘린더를 고려했다.
- 모든 변경 이력에 대한 기록이 추가된다면 타임라인으로 충분히 가치를 제공할 수 있을 것 같다.
- 보고싶은 식물을 필터링 해서 볼 수 있다.
- 많은 변경 이력을 캘린더로 관리하기에는 타임라인의 이점이 더 커보인다.

식물 전체에 대해 정보가 필요한 이유는 미래의 정보가 필요하기 때문이라는 생각.
이러한 측면에서 이미 리마인더로 가치를 제공하고있다.

타임라인 고도화로 방향성을 두는것도 좋아보인다.

### 협의 내용
- [반려 식물 단건 히스토리 조회 기능] 생성, 수정, 삭제, 만료 등에 대한 이력을 타임라인에 남긴다.
- 캘린더 대신 타임라인 고도화로 방향성을 정하고 이에 대한 기능을 이야기해본다.

**Comment by WaiNaat** [comment]
### 캘린더의 존재 유무에 대해

- 캘린더가 없으면 내비게이션 바가 허전해 보이지만 그게 이유가 될 순 없다
- 현재 예상되는 캘린더의 기능: 내 모든 식물의 히스토리를 모아본다
- 결국 지금 캘린더가 보여주는 것에 기능적인 특별함이 없다 (단순 타임라인의 또다른 형태에 불과)
- 만약 미래의 일정까지 보여준다면 의미가 있다
  - 미래의 일정도 보여준다면 캘린더에서 미래 일정도 보여주면서 타임라인도 고도화해야한다
  - 미래 일정이라고 해봤자 바로 다음 1회의 물주기
  - 이후에 분갈이와 같은 다른 기능들까지 추가된다면 캘린더의 역할이 특별해질 것
  - 분갈이같은 기능도 타임라인에서 필터링하면 충분히 보여줄 수 있다

**결론: 캘린더 기능을 없애고 타임라인을 고도화하자**

### 타임라인 고도화

- 물주기 완료 (이전 날짜부터 며칠 후에 줬는지도 표시)
- 반려식물 정보 변경 이력
    - 물주기 주기
    - 화분 종류, 채광, 통풍, 위치
    - 별명?
- 등록한 날 / 떠나보낸 날
- 필터링
    - 물을 준 날만 보기
    - 각 이력 하나에 대한 변경 기록만 보기
    - 모두 보기 (default)

### 캘린더의 의미

캘린더: 날짜별로 볼 수 있다
타임라인: 식물 하나에 대한 정보를 볼 수 있다

타임라인에 새로운 기능이 붙는 것이 캘린더를 버리겠다는 가정을 했기 때문일 수 있다.
캘린더를 하지 않기로 마음을 먹었기 때문에 타임라인에 타임라인의 역할이 아닌 것들이 추가되는 게 아닌가?

### 다시, 캘린더 기능을 해야 하나?

투표 결과 3:4, 캘린더는 낮은 우선순위로
